api_confluence.py

```diff
@@ -209,8 +239,169 @@ def pages():
             page_payload_write(page_id, payload)
 
 
+# ------------------------------------------------------------------
+# Page rendering: Custom pages (XML)
+# ------------------------------------------------------------------
+
+
+def render_xml_template(template_path, params):
+    """ 1. open Mustache-style (jinja compatible) XML template
+        2. parameterize template with params """
+
+    with open(template_path, 'r') as file:
+        template_content = file.read()
+
+    template = Template(template_content)
+    rendered_xml = template.render(**params)
+
+    return rendered_xml
+
+
+# UTILITY FUNCTIONS
+
+def page_content_retrieve_xml(page_id):
+    """Utility function only
+
+    Given page_id, write XML content to a local XML template file
+    """
+    page_url = url_page_content_storage(page_id)
+    resp = page_object(page_url)
+
+    soup = BeautifulSoup(resp, "html.parser")
+    pretty_xml = soup.prettify()
+    with open("confluence_page_content.xml", "w", encoding="utf-8") as f:
+        f.write(pretty_xml)
+
+
+def page_content_insert_xml(page_id, params):
+    """Utility function only
+
+    NOTE: Confluence API returns page content in XML format.
+
+    Use this function as a stand-alone utility / example for pulling page
+    content for rewriting pages or creating new XML template configs.
+    """
+
+    TEMPLATE_PATH = f"{PATH_XML_FILES}/build-validation.xml"
+
+    # XML template
+    # with open(TEMPLATE_PATH, "r", encoding="utf-8") as file:
+    #     xml_content = file.read()
+    output_xml = render_xml_template(TEMPLATE_PATH, params)
+
+    # Confluence page (for updating)
+    page_url = url_page(page_id)
+
+    # Write output_xml to page
+    print(f"UPDATING PAGE - page_id: {page_id}")
+    page_data = page_object(page_url)
+    current_version = page_data["version"]["number"]
+
+    payload = page_payload(page_id, page_title, page_data, current_version, output_xml) # noqa
+    page_payload_write(page_id, payload)
+
+
+# REPORT: Build Validation
```

Comment:
Unlike the Looker Graph pages which all have the same format and config file format, for any other (custom) page, we will likely need to craft data for insertion first in the `api_confluence.py` module.  Hence, this section with functions specifically for creating this specific report.